src/dbusgateway/dbusgateway.cc

```diff
@@ -1,14 +1,15 @@
 #include "dbusgateway.h"
 
 #include <boost/shared_ptr.hpp>
+#include "dbusutilities.h"
 #include "logger.h"
 #include "types.h"
 
 DbusGateway::DbusGateway(const Config& config_in, command::Channel* command_channel_in)
     : stop(false), config(config_in), command_channel(command_channel_in), swlm(config) {
   dbus_threads_init_default();
   dbus_error_init(&err);
-  conn = dbus_bus_get(DBUS_BUS_SESSION, &err);
+  conn = connectToDbus(config_in.dbus, &err);
```

Comment:
We'll have to do some conditional includes because dbus is only needed when BUILD_GENIVI is defined, otherwise should be fine